src/webgldrawer.js

```diff
@@ -887,51 +887,63 @@
+
+                            let left = (tile.x === 0 ? 0 : overlapFraction.x) * sourceWidthFraction;
+                            let top = (tile.y === 0 ? 0 : overlapFraction.y) * sourceHeightFraction;
+                            let right = (tile.isRightMost ? 1 : 1 - overlapFraction.x) * sourceWidthFraction;
+                            let bottom = (tile.isBottomMost ? 1 : 1 - overlapFraction.y) * sourceHeightFraction;
+                            position = this._makeQuadVertexBuffer(left, right, top, bottom);
+                        } else if (sourceWidthFraction === 1 && sourceHeightFraction === 1) {
+                            // no overlap and no padding: this texture can use the unit quad as its position data
+                            position = this._unitQuad;
+                        } else {
+                            position = this._makeQuadVertexBuffer(0, sourceWidthFraction, 0, sourceHeightFraction);
+                        }
 
-                gl.activeTexture(gl.TEXTURE0);
-                gl.bindTexture(gl.TEXTURE_2D, texture);
-                // Set the parameters so we can render any size image.
-                gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
-                gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
-                gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, this._textureFilter());
-                gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, this._textureFilter());
-
-                try{
-                    // This depends on gl.TEXTURE_2D being bound to the texture
-                    // associated with this canvas before calling this function
-                    gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, data);
-                } catch (e){
-                    $.console.error('Error uploading image data to WebGL', e);
+                        gl.activeTexture(gl.TEXTURE0);
+                        gl.bindTexture(gl.TEXTURE_2D, texture);
+                        // Set the parameters so we can render any size image.
+                        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+                        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+                        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, this._textureFilter());
+                        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, this._textureFilter());
+
+                        try {
+                            // This depends on gl.TEXTURE_2D being bound to the texture
+                            // associated with this canvas before calling this function
+                            gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, data);
+                        } catch (e){
+                            // Todo a bit dirty re-use of the tainted flag, but makes the code more stable
```

Comment:
`isTainted()` is not correctly named, since it reflects only one of many causes why we are not able to render data. Here, we catch an exception and we would like to fall back to different renderer, but we need to set `isTainted` flag which might be misleading. Related to misunderstandings in https://github.com/openseadragon/openseadragon/issues/2623

Reply:
Sounds like we should add this to the list of things to address after the merge.